packages/astro/src/assets/services/squoosh.ts

```diff
@@ -28,6 +30,28 @@ const qualityTable: Record<
 	// Squoosh's PNG encoder does not support a quality setting, so we can skip that here
 };
 
+async function getRotationForEXIF(
+	transform: LocalImageTransform,
+	inputBuffer: Buffer
+): Promise<Operation | undefined> {
+	// check EXIF orientation data and rotate the image if needed
+	const meta = await imageMetadata(transform.src, inputBuffer);
+
+	if (!meta) return undefined;
+
+	switch (meta.orientation) {
+		case 3:
+		case 4:
+			return { type: 'rotate', numRotations: 2 };
+		case 5:
+		case 6:
+			return { type: 'rotate', numRotations: 1 };
+		case 7:
+		case 8:
+			return { type: 'rotate', numRotations: 3 };
+	}
```

Comment:
Would be nice to link to https://exiftool.org/TagNames/EXIF.html for an explanation of these magic numbers. Or an inline comment of what each number represent.

Reply:
Pff, imagine not knowing the EXIF rotation numbers by heart 

(I agree, will update, thank you!)